stats/stats-server/src/server.rs

```diff
@@ -56,6 +59,106 @@ fn grpc_router<S: StatsService>(
         .add_service(StatsServiceServer::from_arc(stats))
 }
 
+fn is_threshold_passed(
+    threshold: &ToggleableThreshold,
+    float_value: Option<String>,
+    value_name: &str,
+) -> Result<bool, anyhow::Error> {
+    let threshold = if threshold.enabled {
+        threshold.threshold
+    } else {
+        return Ok(true);
+    };
+    let value = float_value
+        .map(|s| s.parse::<f64>())
+        .transpose()
+        .context(format!("Parsing `{value_name}`"))?;
+    let Some(value) = value else {
+        anyhow::bail!("Received `null` value of `{value_name}`. Can't determine indexing status.",);
+    };
+    if value < threshold {
+        info!(
+            threshold = threshold,
+            current_value = value,
+            "Threshold for `{value_name}` is not satisfied"
+        );
+        Ok(false)
+    } else {
+        info!(
+            threshold = threshold,
+            current_value = value,
+            "Threshold for `{value_name}` is satisfied"
+        );
+        Ok(true)
+    }
+}
+
+fn is_retryable_code(status_code: &reqwest::StatusCode) -> bool {
+    matches!(
+        *status_code,
+        StatusCode::INTERNAL_SERVER_ERROR
+            | StatusCode::SERVICE_UNAVAILABLE
+            | StatusCode::GATEWAY_TIMEOUT
+            | StatusCode::TOO_MANY_REQUESTS
+            | StatusCode::IM_A_TEAPOT
+    )
+}
+
+async fn wait_for_blockscout_indexing(
+    api_config: blockscout_client::Configuration,
+    wait_config: StartConditionSettings,
+) -> Result<(), anyhow::Error> {
+    loop {
+        match blockscout_client::apis::main_page_api::get_indexing_status(&api_config).await {
+            Ok(result)
+                if is_threshold_passed(
+                    &wait_config.blocks_ratio,
+                    result.indexed_blocks_ratio.clone(),
+                    "indexed_blocks_ratio",
+                )? && is_threshold_passed(
+                    &wait_config.internal_transactions_ratio,
+                    result.indexed_internal_transactions_ratio.clone(),
+                    "indexed_internal_transactions_ratio",
+                )? =>
+            {
```

Comment:
```suggestion
            Ok(result)
                if is_threshold_passed(
                    &wait_config.blocks_ratio,
                    result.indexed_blocks_ratio.clone(),
                    "indexed_blocks_ratio",
                ).context("check index block ratio")? && is_threshold_passed(
                    &wait_config.internal_transactions_ratio,
                    result.indexed_internal_transactions_ratio.clone(),
                    "indexed_internal_transactions_ratio",
                ).context("check internal transaction index")? =>
            {
```

Reply:
This name is supplied to errors from the 3rd argument anyway, so I see no reason to repeat it.
One could replace it with this context but then we need to figure out how to pass it to logs (spans? sounds kinda repetitive to add)